midealocal/devices/da/__init__.py

```diff
@@ -137,30 +137,39 @@ def process_message(self, msg: bytes) -> dict[str, Any]:
         ]
         for status in self._attributes:
             if hasattr(message, str(status)):
+                value = getattr(message, str(status))
                 if status == DeviceAttributes.progress:
-                    self._attributes[status] = progress[getattr(message, str(status))]
+                    self._attributes[status] = (
+                        None if value >= len(progress) else progress[value]
+                    )
                 elif status == DeviceAttributes.program:
-                    self._attributes[status] = program[getattr(message, str(status))]
+                    self._attributes[status] = (
+                        None if value >= len(program) else program[value]
+                    )
                 elif status == DeviceAttributes.rinse_level:
-                    temp_rinse_level = getattr(message, str(status))
+                    temp_rinse_level = value
                     if temp_rinse_level == MIN_TEMP:
-                        self._attributes[status] = "-"
+                        self._attributes[status] = None
                     else:
                         self._attributes[status] = temp_rinse_level
                 elif status == DeviceAttributes.dehydration_speed:
-                    temp_speed = getattr(message, str(status))
-                    if temp_speed == MIN_TEMP:
-                        self._attributes[status] = "-"
-                    else:
-                        self._attributes[status] = speed[temp_speed]
+                    self._attributes[status] = (
+                        "-" if value >= len(speed) else speed[value]
+                    )
                 elif status == DeviceAttributes.detergent:
-                    self._attributes[status] = detergent[getattr(message, str(status))]
+                    self._attributes[status] = (
+                        None if value >= len(detergent) else detergent[value]
+                    )
                 elif status == DeviceAttributes.softener:
-                    self._attributes[status] = softener[getattr(message, str(status))]
+                    self._attributes[status] = (
+                        None if value >= len(softener) else softener[value]
+                    )
                 elif status == DeviceAttributes.wash_strength:
-                    self._attributes[status] = strength[getattr(message, str(status))]
+                    self._attributes[status] = (
+                        "-" if value >= len(strength) else strength[value]
```

Comment:
None ?

Reply:
The first item of the array is "-". Thats why I left - instead of None.